sdk/tables/data-tables/src/TableClient.ts

```diff
@@ -556,7 +564,11 @@ export class TableClient {
   ): Promise<SetAccessPolicyResponse> {
     const { span, updatedOptions } = createSpan("TableClient-setAccessPolicy", options);
     try {
-      return this.table.setAccessPolicy(this.tableName, { ...updatedOptions, tableAcl });
+      const serlializedAcl = serializeSignedIdentifiers(tableAcl);
+      return this.table.setAccessPolicy(this.tableName, {
```

Comment:
Yes, we should `await` if we want try/catch to fire on `this.table.setAccessPolicy` rejecting, otherwise it won't:

```ts
async function foo() {
    try {
        return doSomethingAsync();
    } catch(e) {
        console.log("this won't work");
    }
}

async function bar() {
    try {
        return await doSomethingAsync();
    } catch(e) {
        console.log("this will work the way you want it to!");
    }
}

function doSomethingAsync() {
   return Promise.reject(new Error("oh no!"));
}
```

Reply:
This is a great catch! I'm updating this and also awaiting a few others we were missing